src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -578,19 +578,32 @@ def update_iot_hub_custom(instance,
     if fileupload_notification_ttl is not None:
         ttl = timedelta(hours=fileupload_notification_ttl)
         instance.properties.messaging_endpoints['fileNotifications'].ttl_as_iso8601 = ttl
-    # if setting a fileupload storage identity or changing fileupload to identity-based
-    if fileupload_storage_identity or fileupload_storage_authentication_type == AuthenticationType.IdentityBased.value:
-        _validate_fileupload_identity(instance, fileupload_storage_identity)
-
-    default_storage_endpoint = _process_fileupload_args(
-        instance.properties.storage_endpoints['$default'],
-        fileupload_storage_connectionstring,
-        fileupload_storage_container_name,
-        fileupload_sas_ttl,
-        fileupload_storage_authentication_type,
-        fileupload_storage_container_uri,
-        fileupload_storage_identity,
-    )
+    # only bother with $default storage endpoint checking if modifying fileupload params
+    if any([
+            fileupload_storage_connectionstring, fileupload_storage_container_name, fileupload_sas_ttl,
+            fileupload_storage_authentication_type, fileupload_storage_container_uri, fileupload_storage_identity]):
+        default_storage_endpoint = instance.properties.storage_endpoints.get('$default', {})
+        # no default storage endpoint, either recreate with existing params or throw an error
+        if not default_storage_endpoint:
+            if not fileupload_storage_connectionstring and not fileupload_storage_container_name:
```

Comment:
if not any([fileupload_storage_connectionstring, fileupload_storage_container_name])

Reply:
changed to `if not all` - if either of these values aren't set the error should be raised.